### DISCUSSION TITLE
GdUnit3 Beta 0.9.0 Feedback

### DISCUSSION THREAD

**Discussion by MikeSchulze** [MikeSchulze/gdUnit3#3] [General]
Hi Godot users,

you are welcome to give me a feedback to the GdUnit3 beta.

Please describe what you like and not like ;)
If you found some bugs than please report this in the issue section of GitHib.
Feel also free to create feature request.

Thanks in advance
Mike

**Comment by clemens-tolboom** [comment]
Just fetched from asset store (0.9.1-beta) but switched back to using the `git checkout` for testing

- awesome 'make test workflow'
- PR #43
- where is a demo?
  - I learned the tests from addon are a source for tests like `res://addons/gdUnit3/test/asserts/GdUnitArrayAssertImplTest.gd`
  - I would like to have an example outside `addons`
- created snake_case file (`standalone_script.gd`with CamelCase `class_name StandaloneScript` (Godot code style convention)
  - the testfile is located in `res://test/standalone_scriptTest.gd` nice
  - content is `class_name standalone_scriptTest` ... I prefer `StandaloneScriptTest`
  - I learned jumping back to file by CTRL-click on name `const __source = 'res://standalone_script.gd'` so no big deal to find file and 'Search file' is no problem either.
- Removing `assert_not_yet_implemented()` without an assert **should** fail but does not ... 'you forgot to add an assert' hint would help
```
func test_add() -> void:
	# remove this line and complete your test
	pass
```
- I ran into not being able to save (test)file. #44
![gdUnit-5-modified](https://user-images.githubusercontent.com/371014/113483947-2b7c9900-94a6-11eb-8690-de4c591aaf58.png)
- I wonder how to test a scene? Esp. `_process()`, `_physics_process`

**Reply by MikeSchulze** [reply]
Hi thanks for your feedback 👍 

`where is a demo?`
Currently no 'demo' exists i will provide later an instruction video like a demo.
But you can follow the wiki for your first steps with GdUnit3
https://github.com/MikeSchulze/gdUnit3/wiki/FirstStep

`I would like to have an example outside addons`
I will provide later more extended examples on the wiki

`content is class_name standalone_scriptTest ... I prefer StandaloneScriptTest`
the name is just created fom the original source name :)
Feel free to create a feature request, i can introduce a setting where converts the class name into camelcase 


`Removing assert_not_yet_implemented() without an assert should fail but does not ... 'you forgot to add an assert' hint would help`

No an empty test will end with success, only the provided asserts will fail the test if the validation fails


`I ran into not being able to save (test)file. #44`
Thanks for reporting this, i will have a look into

`I wonder how to test a scene? Esp. _process(), _physics_process`
Yes is current not possibe, it will be supported in future see #31

**Reply by clemens-tolboom** [reply]
I'll create (and should have added) seperate feedback to not drown in the text.

**Comment by clemens-tolboom** [comment]
## Demo

No video/wiki please. I like this short 3 lines https://github.com/MikeSchulze/gdUnit3/wiki#example but it would be great to have some working examples in the repo ... `.gitattributes` gives the opportunity to have runnable examples without cluttering Addons downloads. That allows people to clone the repo, run the demo's and fiddle with them. I know this is opionates so feel free to ignore :-p

**Reply by MikeSchulze** [reply]
I got the point, yes this makes sense. I will add some examples later.

**Comment by clemens-tolboom** [comment]
## Empty test

Guess I have to learn not to remove `assert_not_yet_implemented()` too early when doing TDD which is doable ... I added `return` to temp skip a test taking long.
